blocks/math/include/gnuradio-4.0/math/Math.hpp

```diff
@@ -115,6 +125,221 @@ using Multiply = MathOpMultiPortImpl<T, std::multiplies<T>>;
+GR_REGISTER_BLOCK("gr::blocks::math::Not", gr::blocks::math::MathOpImpl, ([T], std::bit_not<[T]>), [ uint8_t, uint16_t, uint32_t, uint64_t, int8_t, int16_t, int32_t, int64_t, float, double, gr::UncertainValue<float>, gr::UncertainValue<double>, std::complex<float>, std::complex<double> ])
+GR_REGISTER_BLOCK("gr::blocks::math::Abs", gr::blocks::math::MathOpImpl, ([T], std::abs_op<[T]>), [ uint8_t, uint16_t, uint32_t, uint64_t, int8_t, int16_t, int32_t, int64_t, float, double ])
+
+template<typename T, typename op>
+requires(std::is_arithmetic_v<T>)
+struct MathOpSinglePortImpl : public gr::Block<MathOpSinglePortImpl<T, op>> {
+    using Description = Doc<R""(
+    @brief Math block transforming a single input to a single output with a given operation
+
+    Depending on the operator op this block computes:
+    - Negate: out = - in
+    - Not: out = ~ in
+    - Abs: out = abs(in)
+    )"">;
+
+    // ports
+    PortIn<T>  in;
+    PortOut<T> out;
+
+    GR_MAKE_REFLECTABLE(MathOpSinglePortImpl, in, out);
+
+    [[nodiscard]] constexpr T processOne(const auto& a) const noexcept { return op{}(a); }
+};
+
+template<typename T>
+using Negate = MathOpSinglePortImpl<T, std::negate<T>>;
+template<typename T>
+using Not = MathOpSinglePortImpl<T, std::bit_not<T>>;
+
+template<class T = void>
+struct abs_op {
+    constexpr T operator()(const T& v) const {
+        if constexpr (std::is_unsigned_v<T>) {
+            // |v| == v  ➜ avoid ambiguous std::abs and silence -Wabsolute-value
+            return v;
+        } else if constexpr (std::is_integral_v<T>) {
+            // std::abs promotes to int/long long; cast back to T
+            using Promoted = std::conditional_t<(sizeof(long long) > sizeof(int) &&
+                                                 sizeof(T) > sizeof(int)),
+                                                long long,
+                                                int>;
+            return static_cast<T>(std::abs(static_cast<Promoted>(v)));
+        } else {
+            // float, double, long double: return type is already correct
+            return std::abs(v);
+        }
+    }
+};
+
+template<typename T>
+using Abs = MathOpSinglePortImpl<T, abs_op<T>>;
+
+/* ------------------------------------------------------------------ *
+ *  Integrate : running sum over N samples, output 1 value, decimate   *
+ * ------------------------------------------------------------------ */
+
+ GR_REGISTER_BLOCK("gr::blocks::math::Integrate",
+                  gr::blocks::math::Integrate,
+                  ([T]),
+                  [ uint8_t, uint16_t, uint32_t, uint64_t,
+                    int8_t,  int16_t,  int32_t,  int64_t,
+                    float, double ])
```

Comment:
Can we add complex types here as well.  The GR block supports `std::complex<float>`